node/src/components/block_synchronizer/global_state_synchronizer.rs

```diff
@@ -505,3 +506,571 @@ where
+        utils,
+    };
+    use casper_types::{bytesrepr::Bytes, testing::TestRng};
+    use futures::channel::oneshot;
+    use rand::Rng;
+
+    /// Event for the mock reactor.
+    #[derive(Debug)]
+    enum ReactorEvent {
+        TrieAccumulatorRequest(TrieAccumulatorRequest),
+        ContractRuntimeRequest(ContractRuntimeRequest),
+    }
+
+    impl From<ContractRuntimeRequest> for ReactorEvent {
+        fn from(req: ContractRuntimeRequest) -> ReactorEvent {
+            ReactorEvent::ContractRuntimeRequest(req)
+        }
+    }
+
+    impl From<TrieAccumulatorRequest> for ReactorEvent {
+        fn from(req: TrieAccumulatorRequest) -> ReactorEvent {
+            ReactorEvent::TrieAccumulatorRequest(req)
+        }
+    }
+
+    struct MockReactor {
+        scheduler: &'static Scheduler<ReactorEvent>,
+        effect_builder: EffectBuilder<ReactorEvent>,
+    }
+
+    impl MockReactor {
+        fn new() -> Self {
+            let scheduler = utils::leak(Scheduler::new(QueueKind::weights()));
+            let event_queue_handle = EventQueueHandle::without_shutdown(scheduler);
+            let effect_builder = EffectBuilder::new(event_queue_handle);
+            MockReactor {
+                scheduler,
+                effect_builder,
+            }
+        }
+
+        fn effect_builder(&self) -> EffectBuilder<ReactorEvent> {
+            self.effect_builder
+        }
+
+        async fn expect_trie_accumulator_request(&self, hash: &Digest, peers: &HashSet<NodeId>) {
+            let ((_ancestor, reactor_event), _) = self.scheduler.pop().await;
+            match reactor_event {
+                ReactorEvent::TrieAccumulatorRequest(request) => {
+                    assert_eq!(request.hash, *hash);
+                    for peer in request.peers.iter() {
+                        assert!(peers.contains(peer));
+                    }
+                }
+                _ => {
+                    unreachable!();
+                }
+            };
+        }
+
+        async fn expect_put_trie_request(&self, trie: &TrieRaw) {
+            let ((_ancestor, reactor_event), _) = self.scheduler.pop().await;
+            match reactor_event {
+                ReactorEvent::ContractRuntimeRequest(ContractRuntimeRequest::PutTrie {
+                    trie_bytes,
+                    responder: _,
+                }) => {
+                    assert_eq!(trie_bytes, *trie);
+                }
+                _ => {
+                    unreachable!();
+                }
+            };
+        }
+    }
+
+    fn random_test_trie(rng: &mut TestRng) -> TrieRaw {
+        let data: Vec<u8> = (0..64).into_iter().map(|_| rng.gen()).collect();
+        TrieRaw::new(Bytes::from(data))
+    }
+
+    fn random_sync_global_state_request(
+        rng: &mut TestRng,
+        num_random_peers: usize,
+        responder: Responder<Result<Digest, Error>>,
+    ) -> (SyncGlobalStateRequest, TrieRaw) {
+        let block = Block::random(rng); // Random block
```

Comment:
Nit:

Excessive comment.

Reply:
Removed redundant comment in https://github.com/casper-network/casper-node/pull/3568/commits/9cf57e53b38f6c80dea20521504ba8b3b8c93ce8